crates/sui-indexer/migrations_v2/2023-08-19-044044_checkpoints/up.sql

```diff
@@ -0,0 +1,26 @@
+CREATE TABLE checkpoints
+(
+    sequence_number                     bigint       PRIMARY KEY,
+    checkpoint_digest                   bytea        NOT NULL,
+    epoch                               bigint       NOT NULL,
+    -- total transactions in the network at the end of this checkpoint (including itself)
+    network_total_transactions          bigint       NOT NULL,
+    previous_checkpoint_digest          bytea,
+    -- if this checkpoitn is the last checkpoint of an epoch
+    end_of_epoch                        boolean      NOT NULL,
```

Comment:
In the GraphQL schema we put the end of epoch data on the last checkpoint in the epoch.  IIRC I did it this way because that's how it shows up on the protocol side, but in the data layer it looks like the end of epoch data resides on the epoch when it has ended.

Is that right?  If so, I'm tempted to get rid of `EndOfEpochData` on the checkpoint, and keep it just on `Epoch`.  If we do that, I don't think we need the `end_of_epoch` boolean either, because we can figure that out from the checkpoint's epoch instead (i.e. keep track of the epoch's first and last checkpoint instead of tracking whether each checkpoint is the last checkpoint of an epoch).

Reply:
>  IIRC I did it this way because that's how it shows up on the protocol side

you are right, this is what we are talking about in slack too. We are re-designing the data layer together so whatever makes the most sense will be in the new data layer. No constraint on this end.

> I'm tempted to get rid of EndOfEpochData on the checkpoint
I don't think it's worthwhile changing core structure if it's what you mean. Again on the data layer we can present it in the most desired way.

> I don't think we need the end_of_epoch boolean either, because we can figure that out from the checkpoint's epoch instead (i.e. keep track of the epoch's first and last checkpoint instead of tracking whether each checkpoint is the last checkpoint of an epoch).

When client queries a checkpoint, without this `end_of_epoch` boolean, you need to join it with `epoch` table. A boolean field is small enough that IMO is not worth the join

Reply:
`end_of_epoch_data` is part of existing rpc `Checkpoint` [struct](https://github.com/MystenLabs/sui/blob/main/crates/sui-json-rpc-types/src/sui_checkpoint.rs#L55). Is it ok to break it by always returning None? Or do we want to keep this column for now and delete it when json rpc is deprecated?

Reply:
I don't think we want to break the existing RPC, but we don't have to right?  Even if we changed the data model, the JSON-RPC endpoint could still construct an `EndOfEpochData` by querying its epoch and its next epoch, which is not very efficient, but lets us build the new data model while still supporting the old query pattern.

I'm happy with either approach (keep `EndOfEpochData` in the data layer, and shim it in GraphQL until JSON-RPC is deprecated, or remove `EndOfEpochData` in the data layer, and shim the access in JSON-RPC).

Reply:
> lets us build the new data model while still supporting the old query pattern.

I wouldn't worry too much about having a couple of more columns now. It's very easy to drop them when they are no long needed. But I don't feel strongly about this one either.

Reply:
and now it's updated